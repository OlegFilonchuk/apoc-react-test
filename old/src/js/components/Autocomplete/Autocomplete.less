.autocomplete {
  .sc-dropdown-menus {
    width: 100%;
    left: 0;
    top: 31px;

    &:empty {
      display: none;
    }
  }
}
