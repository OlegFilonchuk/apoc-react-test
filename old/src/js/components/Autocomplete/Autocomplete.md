## Simple Autocomplete

    const options = [
      { label: 'first', value: 'first' },
      { label: 'second', value: 'second' },
      { label: 'third', value: 'third' }
    ];

    const value = "third";

    const initialState = {
      options,
      value
    };

    function autocompleteChange(value) {
      setState({ value });
    }

    (
      <Autocomplete
        name="simple-example-autocomplete"
        options={state.options}
        value={state.value}
        onChange={autocompleteChange}
      />
    )

## Simple Autocomplete with empty options list
    const options = [];

    const initialState = {
      options
    };

    (
      <Autocomplete name="empty-result-dropdown" options={state.list} />
    )
