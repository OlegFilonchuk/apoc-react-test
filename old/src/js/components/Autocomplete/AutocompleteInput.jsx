import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Input from '../Input/Input';

import './AutocompleteInput.less';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

const BASE_CLASS = 'autocomplete-input';

export default class AutocompleteInput extends Component {
  onInputClick = e => {
    this.props.onClick(e, this.expandedValue);
    this.props.showFilteredOptions(this.props.value);
  };

  onButtonClick = e => {
    if (!this.props.disabled) {
      this.props.onClick(e, this.expandedValue);
      this.props.showAllOptions();
    }
  };

  onInputRef = ref => {
    this.root = ref;
  };

  get expandedValue() {
    return !(this.root !== document.activeElement && this.props.isExpanded);
  }

  get errors() {
    return this.props.children ? <div className="errors">{this.props.children}</div> : null;
  }

  get wrapperClass() {
    return classNames({
      disabled: this.props.disabled,
      [BASE_CLASS]: true,
      'input-wrapper': true,
      error: this.props.hasError
    });
  }

  get arrowClass() {
    return this.expandedValue ? 'sc-icon-down-arrow' : 'sc-icon-up-arrow';
  }

  handleChange = e => this.props.onChange(e.target.value);

  render() {
    const { placeholder, lettersLeftCount, value, maxLength, hasError, disabled } = this.props;

    return (
      <div className={this.wrapperClass}>
        <div className={`${BASE_CLASS}__wrapper`}>
          <Input
            onInputRef={this.onInputRef}
            onClick={this.onInputClick}
            onChange={this.handleChange}
            value={value}
            placeholder={placeholder}
            lettersLeftCount={lettersLeftCount}
            maxLength={maxLength}
            hasError={hasError}
            disabled={disabled}
            {...getDataTestElementProps(this.props)}
          />
        </div>
        <button className={`${BASE_CLASS}__button`} onClick={this.onButtonClick}>
          <span className={this.arrowClass} />
        </button>
        {this.errors}
      </div>
    );
  }
}

AutocompleteInput.propTypes = {
  /**
   * Input value
   */
  value: PropTypes.string,

  /**
   * Is options list expanded
   */
  isExpanded: PropTypes.bool,

  /**
   * onClick callback that is called to show/hide suggestions
   */
  onClick: PropTypes.func,

  /**
   * onChange callback
   */
  onChange: PropTypes.func,

  /**
   * Input placeholder
   */
  placeholder: PropTypes.string,

  /**
   * If input is with characters limit this prop need to be passed to Input component
   */
  lettersLeftCount: PropTypes.number,

  /**
   * Defines maximum characters length
   */
  maxLength: PropTypes.number,

  /**
   * Flag if input has errors for styling
   */
  hasError: PropTypes.bool,

  /**
   * Errors list
   */
  children: PropTypes.node,

  /**
   * Callback to filter results based on input value
   */
  showFilteredOptions: PropTypes.func,

  /**
   * Callback to show all suggestions
   */
  showAllOptions: PropTypes.func,

  /**
   * Is input disabled
   */
  disabled: PropTypes.bool,

  ...dataTestElementPropTypes
};

AutocompleteInput.defaultProps = {
  value: '',
  isExpanded: false,
  onClick: () => true,
  onChange: () => true,
  placeholder: '',
  lettersLeftCount: null,
  maxLength: null,
  hasError: false,
  children: null,
  showFilteredOptions: () => true,
  showAllOptions: () => true,
  disabled: false
};
