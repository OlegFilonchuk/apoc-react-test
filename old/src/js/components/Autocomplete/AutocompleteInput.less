@import '~chameleon/less/variables.less';

@gray: #ccc;
@button-square-size: 31px;

@disabled-background: #eee;
@disabled-text: #b2b2b2;
@disabled-border: #b2b2b2;

.autocomplete-input {
  &__wrapper {
    display: inline-block;
    width: calc(100% - @button-square-size);

    .input-field__char-limit-counter {
      height: calc(100% - 8px);
      top: 4px;
      right: 4px;
      border: 1px solid @gray;
    }

    input {
      border-right: 0;
    }
  }

  &__button {
    width: @button-square-size;
    height: @button-square-size;
    border: 1px solid @primary;
    display: inline-block;
    float: right;
    text-align: center;
    cursor: pointer;

    &:focus {
      outline: none;
    }

    span {
      margin-top: 3px;
      color: @primary;
    }
  }

  &.error &__button {
    border: 1px solid @error;

    span {
      color: @error;
    }
  }

  &.disabled &__button {
    background: @disabled-background;
    border-color: @disabled-border;
    cursor: not-allowed;

    span {
      color: @disabled-text;
    }
  }
}
