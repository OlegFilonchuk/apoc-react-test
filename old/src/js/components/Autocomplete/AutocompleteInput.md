Purpose of this component is to be used together with Autocomplete.

## Standalone example

    <AutocompleteInput
      onChange={() => { console.log('Value changed') }}
      onClick={() => { console.log('Input clicked') }}
      value="some value"
      isExpanded={false}
    />
