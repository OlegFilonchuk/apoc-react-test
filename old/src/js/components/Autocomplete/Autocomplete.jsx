import React from 'react';
import PropTypes from 'prop-types';
import isEqual from 'lodash/isEqual';
import classNames from 'classnames';

import Dropdown from '../Dropdown/Dropdown';
import DropdownOption from '../Dropdown/DropdownOption';
import AutocompleteInput from './AutocompleteInput';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

import './Autocomplete.less';

const BASE_CLASS = 'autocomplete';

class Autocomplete extends React.Component {
  state = {
    options: []
  };

  componentDidMount() {
    this.setPropsOptions(this.props.options);
  }

  componentWillReceiveProps(nextProps) {
    if (!isEqual(this.props.options, nextProps.options)) {
      this.setPropsOptions(nextProps.options);
    }
  }

  onChange = value => {
    this.props.onChange(value);
    this.showFilteredOptions(value);
  };

  setPropsOptions = options => {
    this.setState({
      options
    });
  };

  getFilteredOptions = value =>
    this.props.options
      .filter(option => option.label.toLowerCase().startsWith(value.toLowerCase()))
      .map(({ label, value: optionValue }) => ({ label, value: optionValue }));

  get options() {
    return this.state.options.length
      ? this.state.options.map(({ label, value }) => (
          <DropdownOption
            key={`${value}-${label}`}
            label={label}
            value={value}
            selected={value === this.selectedOption}
          />
        ))
      : [];
  }

  get expanded() {
    return this.state.options.length ? this.props.isExpanded : false;
  }

  get selectedOption() {
    const selectedOption = this.state.options.find(option => option.value === this.props.value);

    return selectedOption ? selectedOption.value : null;
  }

  showFilteredOptions = value => {
    this.setState({
      options: this.getFilteredOptions(value)
    });
  };

  showAllOptions = () => {
    this.setState({
      options: this.props.options
    });
  };

  handleOptionChange = option => this.onChange(option.value);

  render() {
    const { placeholder, lettersLeftCount, value, maxLength, hasError, disabled, children, className } = this.props;
    const finalClassName = classNames(BASE_CLASS, className);

    return (
      <div className={finalClassName}>
        <Dropdown
          name={name}
          onChange={this.handleOptionChange}
          value={value}
          button={
            <AutocompleteInput
              onChange={this.onChange}
              lettersLeftCount={lettersLeftCount}
              value={value}
              isExpanded={this.expanded}
              showAllOptions={this.showAllOptions}
              showFilteredOptions={this.showFilteredOptions}
              maxLength={maxLength}
              hasError={hasError}
              disabled={disabled}
              placeholder={placeholder}
              {...getDataTestElementProps(this.props)}
              id={`${name}-input`}
            >
              {children}
            </AutocompleteInput>
          }
        >
          {this.options.length ? this.options : null}
        </Dropdown>
      </div>
    );
  }
}

Autocomplete.propTypes = {
  /**
   * Input value
   */
  value: PropTypes.string,

  /**
   * Array of options that will be used to show suggestions
   */
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string
    })
  ),

  /**
   * Is input disabled
   */
  disabled: PropTypes.bool,

  /**
   * Function that will be called when input value changes
   */
  onChange: PropTypes.func,

  /**
   * If autocomplete has characters limits this prop needs to be passed to Input component
   */
  lettersLeftCount: PropTypes.number,

  /**
   * Is options list expanded
   */
  isExpanded: PropTypes.bool,

  /**
   * Defines maximum characters length
   */
  maxLength: PropTypes.number,

  /**
   * Flag if input has errors for styling
   */
  hasError: PropTypes.bool,

  /**
   * Errors list
   */
  children: PropTypes.node,

  /**
   * Input placeholder
   */
  placeholder: PropTypes.string,

  /**
   * Any custom class names
   */
  className: PropTypes.string,

  ...dataTestElementPropTypes
};

Autocomplete.defaultProps = {
  value: '',
  options: [],
  disabled: false,
  onChange: value => value,
  lettersLeftCount: null,
  isExpanded: false,
  maxLength: null,
  hasError: false,
  children: null,
  placeholder: null,
  className: ''
};

export default Autocomplete;
