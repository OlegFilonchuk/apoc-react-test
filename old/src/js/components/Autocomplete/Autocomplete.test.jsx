import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Autocomplete from './Autocomplete';
import AutocompleteInput from './AutocompleteInput';

const defaultProps = { name: 'TestAutocomplete' };

const fakeOptions = [
  { label: 'test', value: 'test' },
  { label: 'test2', value: 'test2' }
];

const mountAutocomplete = (props = {}, mountMethod = mount) =>
  mountMethod(<Autocomplete {...defaultProps} {...props} />);

describe('<Autocomplete>', () => {
  it('should render <Autocomplete>', () => {
    expect(toJson(mountAutocomplete({ options: fakeOptions }))).toMatchSnapshot();
  });

  it('should render <Autocomplete> with empty options', () => {
    const options = [];

    expect(toJson(mountAutocomplete({ options }))).toMatchSnapshot();
  });

  it('should render <Autocomplete> with characters limit', () => {
    const fakeLettersLeftCount = 100;

    expect(toJson(mountAutocomplete({ lettersLeftCount: fakeLettersLeftCount }))).toMatchSnapshot();
  });

  it('should render disabled <Autocomplete>', () => {
    expect(toJson(mountAutocomplete({ disabled: true }))).toMatchSnapshot();
  });

  it('should open suggestions list on input click', () => {
    const autocomplete = mountAutocomplete({ options: fakeOptions });

    expect(autocomplete.find('.sc-open').length).toEqual(0);

    autocomplete
      .find(AutocompleteInput)
      .find('input')
      .simulate('click');

    expect(autocomplete.find('.sc-open').length).toEqual(1);
  });

  it('should open suggestions list on button click', () => {
    const autocomplete = mountAutocomplete({ options: fakeOptions });

    expect(autocomplete.find('.sc-open').length).toEqual(0);

    autocomplete
      .find(AutocompleteInput)
      .find('button')
      .simulate('click');

    expect(autocomplete.find('.sc-open').length).toEqual(1);
  });
});
