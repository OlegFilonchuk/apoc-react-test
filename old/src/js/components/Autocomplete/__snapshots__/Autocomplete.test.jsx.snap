// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete> should render <Autocomplete> 1`] = `
<Autocomplete
  className=""
  disabled={false}
  hasError={false}
  isExpanded={false}
  lettersLeftCount={null}
  maxLength={null}
  name="TestAutocomplete"
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "test",
        "value": "test",
      },
      Object {
        "label": "test2",
        "value": "test2",
      },
    ]
  }
  placeholder={null}
  value=""
>
  <div
    className="autocomplete"
  >
    <Dropdown
      button={
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="nodejs"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      value=""
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        >
          <div
            className="autocomplete-input input-wrapper"
          >
            <div
              className="autocomplete-input__wrapper"
            >
              <Input
                className=""
                disabled={false}
                hasError={false}
                lettersLeftCount={null}
                maxLength={null}
                onChange={[Function]}
                onClick={[Function]}
                onInputRef={[Function]}
                onKeyDown={[Function]}
                placeholder={null}
                submitOnEnter={false}
                suffix=""
                type="text"
                value=""
              >
                <span
                  className="input-field-wrapper"
                >
                  <input
                    className="sc-form-ele input-field with-characters-limit-input"
                    disabled={false}
                    maxLength={null}
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    placeholder={null}
                    type="text"
                    value=""
                  />
                  <div
                    className="input-field__char-limit-counter"
                  >
                    <div
                      className="input-field__char-limit-counter__text"
                    />
                  </div>
                </span>
              </Input>
            </div>
            <button
              className="autocomplete-input__button"
              onClick={[Function]}
            >
              <span
                className="sc-icon-down-arrow"
              />
            </button>
          </div>
        </AutocompleteInput>
        <ul
          aria-labelledby="nodejs-input"
          className="sc-dropdown-menus"
        >
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$test-test"
            label="test"
            name="nodejs"
            onClick={[Function]}
            selected={true}
            showPopupOnHover={false}
            value="test"
          >
            <li
              className="dropdown-option "
            >
              <label
                className="sc-focus"
                title={undefined}
              >
                <span
                  className=" icon"
                />
                test
                <input
                  data-test-element="option-test"
                  defaultChecked={true}
                  disabled={false}
                  name="nodejs"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="test"
                />
              </label>
            </li>
          </DropdownOption>
          <DropdownOption
            className=""
            disabled={false}
            icon=""
            key=".$test2-test2"
            label="test2"
            name="nodejs"
            onClick={[Function]}
            selected={false}
            showPopupOnHover={false}
            value="test2"
          >
            <li
              className="dropdown-option "
            >
              <label
                className=""
                title={undefined}
              >
                <span
                  className=" icon"
                />
                test2
                <input
                  data-test-element="option-test2"
                  defaultChecked={false}
                  disabled={false}
                  name="nodejs"
                  onClick={[Function]}
                  readOnly="readOnly"
                  role="menu"
                  type="radio"
                  value="test2"
                />
              </label>
            </li>
          </DropdownOption>
        </ul>
      </div>
    </Dropdown>
  </div>
</Autocomplete>
`;

exports[`<Autocomplete> should render <Autocomplete> with characters limit 1`] = `
<Autocomplete
  className=""
  disabled={false}
  hasError={false}
  isExpanded={false}
  lettersLeftCount={100}
  maxLength={null}
  name="TestAutocomplete"
  onChange={[Function]}
  options={Array []}
  placeholder={null}
  value=""
>
  <div
    className="autocomplete"
  >
    <Dropdown
      button={
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={100}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="nodejs"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      value=""
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={100}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        >
          <div
            className="autocomplete-input input-wrapper"
          >
            <div
              className="autocomplete-input__wrapper"
            >
              <Input
                className=""
                disabled={false}
                hasError={false}
                lettersLeftCount={100}
                maxLength={null}
                onChange={[Function]}
                onClick={[Function]}
                onInputRef={[Function]}
                onKeyDown={[Function]}
                placeholder={null}
                submitOnEnter={false}
                suffix=""
                type="text"
                value=""
              >
                <span
                  className="input-field-wrapper"
                >
                  <input
                    className="sc-form-ele input-field with-characters-limit-input"
                    disabled={false}
                    maxLength={null}
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    placeholder={null}
                    type="text"
                    value=""
                  />
                  <div
                    className="input-field__char-limit-counter"
                  >
                    <div
                      className="input-field__char-limit-counter__text"
                    >
                      100
                    </div>
                  </div>
                </span>
              </Input>
            </div>
            <button
              className="autocomplete-input__button"
              onClick={[Function]}
            >
              <span
                className="sc-icon-down-arrow"
              />
            </button>
          </div>
        </AutocompleteInput>
        <ul
          aria-labelledby="nodejs-input"
          className="sc-dropdown-menus"
        />
      </div>
    </Dropdown>
  </div>
</Autocomplete>
`;

exports[`<Autocomplete> should render <Autocomplete> with empty options 1`] = `
<Autocomplete
  className=""
  disabled={false}
  hasError={false}
  isExpanded={false}
  lettersLeftCount={null}
  maxLength={null}
  name="TestAutocomplete"
  onChange={[Function]}
  options={Array []}
  placeholder={null}
  value=""
>
  <div
    className="autocomplete"
  >
    <Dropdown
      button={
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="nodejs"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      value=""
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <AutocompleteInput
          disabled={false}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        >
          <div
            className="autocomplete-input input-wrapper"
          >
            <div
              className="autocomplete-input__wrapper"
            >
              <Input
                className=""
                disabled={false}
                hasError={false}
                lettersLeftCount={null}
                maxLength={null}
                onChange={[Function]}
                onClick={[Function]}
                onInputRef={[Function]}
                onKeyDown={[Function]}
                placeholder={null}
                submitOnEnter={false}
                suffix=""
                type="text"
                value=""
              >
                <span
                  className="input-field-wrapper"
                >
                  <input
                    className="sc-form-ele input-field with-characters-limit-input"
                    disabled={false}
                    maxLength={null}
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    placeholder={null}
                    type="text"
                    value=""
                  />
                  <div
                    className="input-field__char-limit-counter"
                  >
                    <div
                      className="input-field__char-limit-counter__text"
                    />
                  </div>
                </span>
              </Input>
            </div>
            <button
              className="autocomplete-input__button"
              onClick={[Function]}
            >
              <span
                className="sc-icon-down-arrow"
              />
            </button>
          </div>
        </AutocompleteInput>
        <ul
          aria-labelledby="nodejs-input"
          className="sc-dropdown-menus"
        />
      </div>
    </Dropdown>
  </div>
</Autocomplete>
`;

exports[`<Autocomplete> should render disabled <Autocomplete> 1`] = `
<Autocomplete
  className=""
  disabled={true}
  hasError={false}
  isExpanded={false}
  lettersLeftCount={null}
  maxLength={null}
  name="TestAutocomplete"
  onChange={[Function]}
  options={Array []}
  placeholder={null}
  value=""
>
  <div
    className="autocomplete"
  >
    <Dropdown
      button={
        <AutocompleteInput
          disabled={true}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        />
      }
      className=""
      emptyMessage="No Filtered Data"
      filterable={false}
      fullWidth={false}
      name="nodejs"
      onChange={[Function]}
      recomputeUseButton={false}
      useSubmitOnEnter={false}
      value=""
      withSuffix={Object {}}
    >
      <div
        className="sc-dropdown"
      >
        <AutocompleteInput
          disabled={true}
          hasError={false}
          id="nodejs-input"
          isExpanded={false}
          lettersLeftCount={null}
          maxLength={null}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          placeholder={null}
          showAllOptions={[Function]}
          showFilteredOptions={[Function]}
          value=""
        >
          <div
            className="disabled autocomplete-input input-wrapper"
          >
            <div
              className="autocomplete-input__wrapper"
            >
              <Input
                className=""
                disabled={true}
                hasError={false}
                lettersLeftCount={null}
                maxLength={null}
                onChange={[Function]}
                onClick={[Function]}
                onInputRef={[Function]}
                onKeyDown={[Function]}
                placeholder={null}
                submitOnEnter={false}
                suffix=""
                type="text"
                value=""
              >
                <span
                  className="input-field-wrapper"
                >
                  <input
                    className="sc-form-ele input-field with-characters-limit-input"
                    disabled={true}
                    maxLength={null}
                    onChange={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    placeholder={null}
                    type="text"
                    value=""
                  />
                  <div
                    className="input-field__char-limit-counter"
                  >
                    <div
                      className="input-field__char-limit-counter__text"
                    />
                  </div>
                </span>
              </Input>
            </div>
            <button
              className="autocomplete-input__button"
              onClick={[Function]}
            >
              <span
                className="sc-icon-down-arrow"
              />
            </button>
          </div>
        </AutocompleteInput>
        <ul
          aria-labelledby="nodejs-input"
          className="sc-dropdown-menus"
        />
      </div>
    </Dropdown>
  </div>
</Autocomplete>
`;
